book_app.rb

```diff
@@ -0,0 +1,133 @@
+require_relative 'item'
+require_relative 'label'
+require_relative 'book'
+require_relative 'book_data_saver'
+require_relative 'book_data_loader'
+
+class App
+  def initialize
+    @labels = DataLoader.read_labels
+    @books = DataLoader.read_books(@labels)
+  end
+
+  def create_item
+    puts 'Enter (1) for Book (2) for MusicAlbum (3) for Game :'
+    num = gets.chomp.to_i
+    case num
+    when 1
+      create_book
+    when 2
+      create_music_album
+    when 3
+      create_game
+    when 4
+      create_movie
+    else
+      puts 'Error'
+    end
+  end
+
+  def create_book
+    title = ask_user('Add a label title')
+    color = ask_user('Add a color for the label')
+    label = Label.new(title, color)
+    @labels << label
+    puts 'label created successfully'
+
+    cover_state = ask_cover_state('Add the cover state for the book (1) good (2) bad')
+    publisher = ask_user('Add the publisher for the book')
+    publish_date = ask_publish_date('Add the publish date for the book. Please use the format YYYY-MM-DD.')
+    book = Book.new(cover_state, publisher, publish_date)
+    book.label = label
+    @books << book
+    puts 'book added successfully'
+  end
+
+  def ask_user(prompt)
+    puts prompt
+    gets.chomp
+  end
+
+  def ask_cover_state(prompt)
+    puts prompt
+    cover_state = gets.chomp.to_i
+    if [1, 2].include?(cover_state)
+      cover_state == 1 ? 'good' : 'bad'
+    else
+      puts 'Invalid cover state. Please use good or bad.'
+      ask_cover_state(prompt)
+    end
+  end
```

Comment:
The current code displays a generic error message when the cover state is invalid. It would be more helpful to provide specific instructions to the user, indicating the allowed values and how to enter them correctly.